pkg/kube/client.go

```diff
@@ -561,7 +561,9 @@ func (c *Client) watchUntilReady(timeout time.Duration, info *resource.Info) err
 	// In the future, we might want to add some special logic for types
 	// like Ingress, Volume, etc.
 
-	_, err = watch.Until(timeout, w, func(e watch.Event) (bool, error) {
+	ctx, cancel := watchtools.ContextWithOptionalTimeout(context.Background(), timeout)
+	defer cancel()
+	_, err = watchtools.UntilWithoutRetry(ctx, w, func(e watch.Event) (bool, error) {
```

Comment:
would it be valuable to use `UntilWithSync` instead, or do we want to intentionally close out the connection when we hit the timeout? Based on the docstrings client-go provided, it looks like they'll [remove this endpoint eventually](https://github.com/kubernetes/client-go/blob/3e116d45271ab22b04a40641e7d431990b2349fd/tools/watch/until.go#L53-L57), and they [recommend using `UntilWithSync`](https://github.com/kubernetes/client-go/blob/3e116d45271ab22b04a40641e7d431990b2349fd/tools/watch/until.go#L108-L109).

What do you think?

Reply:
That function signature is very different, I would rather use as close to a drop-in as possible for helm2